packages/dd-trace/src/iitm.js

```diff
@@ -2,8 +2,21 @@
 
 const semver = require('semver')
 const logger = require('./log')
-
+const { moduleLoadStart } = require('./channel-itm')
 if (semver.satisfies(process.versions.node, '^12.20.0 || >=14.13.1')) {
+  const iitm = require('import-in-the-middle/lib/register.js')
+  const originalRegister = iitm.register
+  const newRegister = function (name, namespace, set, specifier) {
```

Comment:
Let's use a hook instead. If we're missing information from the library, we should update it (or vendor it which we plan to do in the future, but it will be a lot more work to do it properly).

Reply:
- [ ] Use hook and shimmer

Reply:
I've tried to use `addHook` but we can't use it to add hook in `import-in-the-middle` because `import-in-the-middle/lib/register.js` is used to call to `addHook` callback method. 
When the library is able to call to `addHook`, the module that we want to hook is already initialized, and it is not going to call to it again.

I've changed it to use `shimmer.wrap` instead of my initial way, but I can't change it to use `addHook`

Reply:
At the end, we don't need specifier, we can calculate it from filename. so we don't need to update `import-in-the-middle` and we don't need to patch it.